### PR TITLE
Final Capstone Project: Rent-Eaze Booking System - Merge Request

### PR DESCRIPTION
**Pull request by lRebornsl**
**Overview**
This pull request is a formal request to merge our development team's cumulative and concerted efforts from the `dev` branch into the `main` branch. Our Final Capstone Project is a sophisticated rental house booking system, designed and tailored following the prototype provided by Murat Korkmaz on Behance, with our unique spin - instead of motorcycles, we facilitate the booking of rental houses.

Our development team, composed of Javier Grau, Manuel Sanchez, and Anthony Vásquez, has meticulously adhered to coding best practices, proper Gitflow, and comprehensive professional documentation throughout the project lifecycle.

[Pull Request from back-end](https://github.com/grauJavier/final-capstone-back-end/pull/55)

**General Requirements Met**
- Zero linter errors across the codebase.
- Strict adherence to Gitflow practices.
- Professional-level documentation and code commenting.

**Project Implementation Details**

**Chosen Theme**
- Our application is built around the concept of booking rental houses, aiming to streamline the process of scheduling a house with an intuitive user interface.

**Core Features Implemented**
- Normal user login system by typing the email and password.
- The navigation panel provides seamless access to:
  - List of houses.
  - House reservation form.
  - Functionality to add new houses to the system.
  - Option to remove their own houses from the system.
- The main page features a listing of houses available for users.
- Detail pages for each house, with an option for users to reserve it.
- A fully responsive design, with tailored user experiences for both mobile and desktop versions.

**Additional Features (Beyond Basic Requirements)**
- The reservation process is enhanced by allowing users to select only a future date.

**Comprehensive Documentation**
- Detailed API documentation has been included to facilitate understanding and potentially scaling the application in the future.

**Useful Links**
- Back-End Documentation: https://renteaze-d1cc8b293660.herokuapp.com/api-docs/index.html
- Full project deployment: https://rent-eaze.onrender.com/

**Merge Request Justification**
With all required features meticulously implemented, we are confident that our project meets the specified criteria. We kindly request the review and approval for the merging of our `dev` branch into `main`.

We eagerly await any feedback and are fully prepared to conduct further modifications if recommended by our esteemed reviewers.

Thank you for your consideration and time.

Warm regards,
@grauJavier 
@Luffytaro22 
@lRebornsl

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import PublicRoute from './components/utils/PublicRoutes'
+import PlaceForm from './components/place_form/PlaceForm.jsx'
+import NavBar from './components/navigation_bar/NavBar'
+import './App.css'
+import ReservationForm from './components/reservation_form/ReservationForm'
+import MyReservations from './components/my_reservations/MyReservations.jsx'
+import PlaceDetails from './components/place_details/PlaceDetails.jsx'
+import MyPlaces from './components/my_places/MyPlaces.jsx'
+
+function App() {
```
Comment (Meltrust) [inline src/App.jsx:18]: - [ ]  Please, kindly refactor this functional component to ES6 arrow function, so your code complies with the latest standards.

Refactoring is easy:

```javascript
const App = () => {
  // ...
}
```

## Explanation

The `function` keyword has been soft deprecated in almost all the software industry in favor of ES6 arrow functions. This means, almost nobody uses it anymore, so much that it's considered a non-best practice.

That way, your code will meet the latest standards. 💪

File: src/components/my_reservations/MyReservations.jsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import deleteReservation from '../../redux/my_reservations/actions/deleteReservation';
+import fetchMyReservations from '../../redux/my_reservations/actions/fetchMyReservations';
+import { selectUserId } from '../../redux/user/userSlice.js';
+import { formatDistanceToNow } from 'date-fns';
+
+function MyReservations() {
```
Comment (Meltrust) [inline src/components/my_reservations/MyReservations.jsx:8]: - [ ]  Please, kindly refactor this functional component to ES6 arrow function, so your code complies with the latest standards.

Refactoring is easy:

```javascript
const MyReservations= () => {
  // ...
}
```

## Explanation

The `function` keyword has been soft deprecated in almost all the software industry in favor of ES6 arrow functions. This means, almost nobody uses it anymore, so much that it's considered a non-best practice.

That way, your code will meet the latest standards. 💪

File: README.md [hunk 3]
```diff
@@ -1 +1,212 @@
...
+
+Install this project with:
+
+- npm install
+
+### Usage
+
+To run the project, execute the following command:
+
+npm start
```
Comment (Meltrust) [inline README.md:115]: - [ ] This command won't work:

image

The correct command to run the frontend project is `npm run dev`.  Kindly correct it.

Comment (Meltrust) [inline README.md:115]: - [ ] This command won't work:

image

The correct command to run the frontend project is `npm run dev`.  Kindly correct it.

File: src/components/place_form/PlaceForm.jsx [hunk 4]
```diff
@@ -0,0 +1,222 @@
+import { useForm } from 'react-hook-form';
+import { useDispatch, useSelector } from 'react-redux';
+import { useEffect } from 'react';
+import ImageUploader from '../utils/UploadFile';
+import fetchCities from '../../redux/reservation_form/actions/fetchCities';
+import { sendPlaces } from '../../redux/places/placesSlice';
+
+const PlaceForm = () => {
```
Comment (Meltrust) [inline src/components/place_form/PlaceForm.jsx:8]: - [ ] When I try to add a new place, I try to upload a new image by pressing this button:

image

However, after selecting any image, this happens:


image

The screen suddenly turns blank and I get a console error:


image

Kindly, fix this bug so your app runs as intended.  If this is not a bug, this means your app requires special procedures/configurations to be run correctly.  If this is the case, kindly add those instructions to your readme.

That way, your users will be able to add new places. 👍

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @lRebornsl,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Tests are passing✔️
-  Nice code organization ✔️ 
-  Good readme ✔️ 
-  Beautiful frontend ✔️ 



#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

File: src/App.jsx (hunk 1 above)
Comment (lRebornsl) [reply src/App.jsx:18]: Hello!
Thanks for your feedback 🙌🏼 
This is solved 😄

File: src/components/my_reservations/MyReservations.jsx (hunk 2 above)
Comment (lRebornsl) [reply src/components/my_reservations/MyReservations.jsx:8]: Hello!
Thanks for your feedback 🙌🏼 
This is solved 😄

File: README.md (hunk 3 above)
Comment (lRebornsl) [reply README.md:115]: Hello!
Thanks for your feedback 🙌🏼 
This is solved 😄

Comment (lRebornsl) [reply README.md:115]: Hello!
Thanks for your feedback 🙌🏼 
This is solved 😄

File: src/components/place_form/PlaceForm.jsx (hunk 4 above)
Comment (lRebornsl) [reply src/components/place_form/PlaceForm.jsx:8]: Hello!
Thanks for your feedback 🙌🏼
This is solved 😄

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @lRebornsl,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!









#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

File: src/App.jsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+import PublicRoute from './components/utils/PublicRoutes'
+import PlaceForm from './components/place_form/PlaceForm.jsx'
+import NavBar from './components/navigation_bar/NavBar'
+import './App.css'
+import ReservationForm from './components/reservation_form/ReservationForm'
+import MyReservations from './components/my_reservations/MyReservations.jsx'
+import PlaceDetails from './components/place_details/PlaceDetails.jsx'
+import MyPlaces from './components/my_places/MyPlaces.jsx'
+
+const App = () => {
```
[the remark this answers is not in the document]
Comment (lRebornsl) [reply src/App.jsx:18]: Hi @Meltrust 👋

That is not our project, I think that you are confusing with another project 😅

Can you review again please? 🙌🏼

Thanks

**Comment by Meltrust** [comment]
> Hi @Meltrust 👋
> 
> That is not our project, I think that you are confusing with another project 😅
> 
> Can you review again please? 🙌🏼
> 
> Thanks

Hello @grauJavier, thanks for your inquiry.  After some investigation, I can conclude you are right. 

Apologies, this was my mistake confusing the projects in this last re-review.

In response to your inquiry, your review will be redone from scratch.

**Review by Meltrust — APPROVED** [review]
#### Hi @lRebornsl,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight

- Readme instructions have been corrected and improved✔️ 
- Adding new places is now correctly functional✔️ 
- Superb job✔️ 



Cheers and Happy coding!👏👏👏


------

**Pull request merged**